### ISSUE TITLE
Quick, 1-click deployment of a PeerTube instance

### ISSUE THREAD

**Issue by DontUseGithub** [Chocobozzz/PeerTube#2375]
It'd be nice to have something like this, not everyone wants to do everything from scratch and follow all those installation steps.

We could use Heroku or something similar but then Heroku would get that money for hosting it. So why not discuss this with Framasoft to create a PeerTube hosting service? Then you'd link to that service in this repo and people who want a fast PeerTube server deployment would click that button.

TL;DR of this issue: to offer a very quick deployment solution via a third-party service or via Framasoft.

**Comment by Chocobozzz** [comment]
At the moment, Framasoft does not have enough resources and time to provide such service. But third party services are free to create a quick deployment solution. It should not be too hard since we provide docker images.

**Comment by rigelk** [comment]
FYI, Yunohost provides 1-click installs, and last time I checked https://app.spacebear.ee/ hosted solutions.

**Comment by DontUseGithub** [comment]
@Chocobozzz I opened this issue for someone from the project to do the talk with some third party services and set everything up and then to bring that "magic button" and add it to the readme file of PeerTube which people would use to quickly deploy a PeerTube server. The more convenient the "deployment" process gets, the more instances we'll see popping up. This is what I'm trying to say. Other "open source youtube alternatives" are doing this, and I don't want PeerTube to stay behind :disappointed: 

@Chocobozzz @rigelk It looks like spacebear.ee is very close to what we'd want but here is what their faq says:

> Can I host PeerTube with your service?
> 
> Yes, Peertube hosting is available upon request by emailing support@spacebear.ee. The pricing is exactly the same as Pixelfed.

I'm not that skilled, so maybe someone from the project can contact them and ask to make this an easy process without the need of emailing the spacebear.ee team? Ideally we'd want to place a button in the readme file that would have a specific link designed to create a peertube instance right away e.g. `https://app.spacebear.ee/start?hosting=peertube` so once a user click this: 
- he'd have to sign up for an account 
- then choose the plan ($5, $10 ...) 
- and then click a button that would start the process of deploying a peertube instance.

Optionally you could ask for a referral link from them, this way you'll get some extra funding for the project every time someone deploys a peertube server via spacebear.ee. It may sound bad, but if you'll specify in the readme file that that's a referral link and provide them a clean link if they don't wanna use that - that'd make it 100% ethical IMO!

**Comment by rigelk** [comment]
> I opened this issue for someone from the project to do the talk with some third party services and set everything up and then to bring that "magic button" and add it to the readme file of PeerTube which people would use to quickly deploy a PeerTube server.

The main issue is that we don't have any contact with providers (the one I cited has never contacted us), and we don't have any feedback about them either.

> Other "open source youtube alternatives" are doing this […]

I would love to see which, and how they are doing it firsthand.

**Comment by DontUseGithub** [comment]
> I would love to see which, and how they are doing it firsthand.

nodetube has a button in their readme.md file called "Deploy to Heroku" which links to: https://heroku.com/deploy?template=https://github.com/mayeaux/nodetube

**Comment by DontUseGithub** [comment]
When a user wants to use [yunohost](https://install-app.yunohost.org/?app=peertube) for an easy deployment of PeerTube, he might get scared by this warning:

> WARNING: PeerTube is NOT officially supported by the YunoHost team yet. Install it at your own risk. The community maintainer has described this app to be in a working state.

Could someone from PeerTube perhaps help/collaborate with them, so that the yunohost team can start offering official support for PeerTube?

Very often I see people complain about how hard it is to deploy a PeerTube server. Hopefully we can somehow improve their experience in this regard.

**Comment by rigelk** [comment]
> Could someone from PeerTube perhaps help/collaborate with them, so that the yunohost team can start offering official support for PeerTube?

Having a package make it in the official packages is *very* hard (commitment on long term, having multiple architectures supported at the highest level of integration, etc.), and we don't have LDAP support yet (blocking us even if we wanted to make it there).
 
> Very often I see people complain about how hard it is to deploy a PeerTube server. Hopefully we can somehow improve their experience in this regard.

To improve their experience, we would need *their* feedback on what made it hard. Please ask them to come and tell us what their struggle with the installation came down to.

**Comment by gwatcha** [comment]
I have a computer engineer bachelor's and I am a linux hobby-ist and it was just a mess for me to get it running on Manjaro.

I spent about 3 hours and then I gave up around the certificates and integrating with nginx. It would help if the docs were written more specifically so I would not have to infer things for certain steps.

I ended up uninstalling because now I am getting new 'node gyp' 
`gyp ERR! stack Error: connect EHOSTUNREACH 2606:4700:10::6814:162e:443
gyp ERR! stack     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)` 
errors in another project with yarn, and I'm trying to figure out how that happened now.

I am not competent enough to install this and I worry for my system so I gave up. So many different programs to integrate it's perplexing... redis, nginx, postgres, peertube. When it came to associating my own domain name the instructions were too general for me. 

I could run it locally but there was an error message popping up at the bottom 'this host has not configured their peertube instance fully' or something like that. 😕

It's a cool software and now I'm going to look into third parties setting it up for me
